gui/default/syncthing/core/syncthingController.js

```diff
@@ -553,10 +569,22 @@ angular.module('syncthing.core')
                 $scope.connectionsTotal = data.total;
 
                 data = data.connections;
+                $scope.recentRemoteAddressCache = {};
```

Comment:
This variable doesn't need to be on the scope.

I was more imagining along the lines of (pseudo code missing parts):

```
var oldCache = $scope.remoteAddressCache;
var newCache = {};
for (id in data) {
    if ($scope.showRemoteAddress && isNotRelayConnection && port !== "0") {
    var newAddress = "http://" + replaceAddressPort(data[id].address, port);
    if (newAddress in oldCache) {
        newCache = oldCache[newAddress];
    } else {
        newCache[newAddress] = $scope.probeAddress(newAddress);
    }
    if (newCache[newAddress] === true) {
        $scope.idToRemoteAddress[id] = newAddress;
    } else {
        $scope.idToRemoteAddress[id] = "";
    }
}
$scope.remoteAddressCache = newCache;
```

This way probeAddress suits a single responsibility principle better (as it just probes, and not "check cache and maybe probe").